core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -171,5 +172,9 @@ object Consumer {
     }
   }
 
+  def committablePartitionedSource[K, V](settings: ConsumerSettings[K, V]):
+    Source[(TopicPartition, Source[CommittableMessage[K, V], Control]), Control] = {
```

Comment:
I think the idea of emitting sources is nice, if it doesn't introduce resource leak traps. I don't know the details of `SubFlow` and if we can use that for a custom stage. Conversion of the groupBy stage to GraphStage is also pending, in progress I think.